web/jobs/jobs.go

```diff
@@ -466,6 +467,94 @@ func cleanJobs(c echo.Context) error {
 	return c.JSON(200, map[string]int{"deleted": len(ups)})
 }
 
+func purgeJobs(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	if err := middlewares.AllowWholeType(c, webpermissions.DELETE, consts.Jobs); err != nil {
+		return err
+	}
+
+	workersParam := c.QueryParam("workers")
+	monthsParam := c.QueryParam("months")
+
+	var months int = 2
+	var err error
+	if monthsParam != "" {
+		months, err = strconv.Atoi(monthsParam)
+		if err != nil {
+			return err
+		}
+	}
+	workers := jobs.GetWorkersNamesList()
+	if workersParam != "" {
+		workers = strings.Split(workersParam, ",")
+	}
+
+	// Step 1: We want to get all the jobs prior to the date parameter.
+	// Jobs returned are the ones we want to remove
+	d := time.Now().AddDate(0, -months, 0)
```

Comment:
[nit] It feels wrong. I'm not sure why exactly, maybe the fact that we are using a negative number of months instead of a duration.